Add options GO_NO_HOST and GO_NO_TARGET to make.bash and copy binaries
into specific directories depending on the target.

Signed-off-by: Christian Stewart <christian@paral.in>

diff --git a/src/make.bash b/src/make.bash
index fbc6f5d..0c1c4d7 100755
--- a/src/make.bash
+++ b/src/make.bash
@@ -161,18 +161,29 @@ mv cmd/dist/dist "$GOTOOLDIR"/dist
 "$GOTOOLDIR"/go_bootstrap clean -i std
 echo
 
-if [ "$GOHOSTARCH" != "$GOARCH" -o "$GOHOSTOS" != "$GOOS" ]; then
-	echo "# Building packages and commands for host, $GOHOSTOS/$GOHOSTARCH."
-	# CC_FOR_TARGET is recorded as the default compiler for the go tool. When building for the host, however,
-	# use the host compiler, CC, from `cmd/dist/dist env` instead.
-	CC=$CC GOOS=$GOHOSTOS GOARCH=$GOHOSTARCH \
-		"$GOTOOLDIR"/go_bootstrap install -ccflags "$GO_CCFLAGS" -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v std
-	echo
+if [ -z "$GO_NO_HOST" ]; then
+
+  echo "# Building packages and commands for host, $GOHOSTOS/$GOHOSTARCH."
+  # CC_FOR_TARGET is recorded as the default compiler for the go tool. When building for the host, however,
+  # use the host compiler, CC, from `cmd/dist/dist env` instead.
+  CC=$CC GOOS=$GOHOSTOS GOARCH=$GOHOSTARCH \
+    "$GOTOOLDIR"/go_bootstrap install -ccflags "$GO_CCFLAGS" -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v std
+  echo
+
+  # xxx buildroot patch, move to host/ subdir
+  mkdir -p ../bin/host/
+  find ../bin/ -type f -maxdepth 1 -exec mv -i {} ../bin/host/ \;
 fi
 
-echo "# Building packages and commands for $GOOS/$GOARCH."
-CC=$CC_FOR_TARGET "$GOTOOLDIR"/go_bootstrap install $GO_FLAGS -ccflags "$GO_CCFLAGS" -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v std
-echo
+if [ -z "$GO_NO_TARGET" ]; then
+  echo "# Building packages and commands for $GOOS/$GOARCH."
+  CC=$CC_FOR_TARGET "$GOTOOLDIR"/go_bootstrap install $GO_FLAGS -ccflags "$GO_CCFLAGS" -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v std
+  echo
+
+  # xxx buildroot patch, move to target subdir
+  mkdir -p ../bin/${GOOS}_${GOARCH}/
+  find ../bin/ -type f -maxdepth 1 -exec mv -i {} ../bin/${GOOS}_${GOARCH}/ \;
+fi
 
 rm -f "$GOTOOLDIR"/go_bootstrap
 
